website/docs/about_code_generation.mdx

```diff
@@ -55,13 +79,133 @@ This includes but is not limited to:
 - some Riverpod features will be available only with code generation.
 
 At the same time, many applications already use code generation with packages such
-as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).  
+as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## Provider type
+
+### <span style={RED_STYLE}>[Sync/Async]</span><span style={BLUE_STYLE}>[Stateless/Stateful]</span>Provider
```

Comment:
We probably want a second section next to this one about:

- Making a provider autoDispose/keepAlive. It should mention that autoDispose is on by default and why.
  It's on by default because this better fits Riverpod's mantra. It was originally off by default for folks migrating from pkg:provider. But theoretically, all providers should be autoDispose.

- Passing parameters to providers (family). It should showcase how to do that, and all parameters are supported. So we can have 2+, we can have named parameters, optional ones, or even default values.